qrm/scan_context.go

```diff
@@ -187,28 +187,30 @@ func (s *ScanContext) getGroupKeyInfo(
 		field := structType.Field(i)
 		fieldType := indirectType(field.Type)
 
-		if !isSimpleModelType(fieldType) {
-			if fieldType.Kind() != reflect.Struct {
-				continue
-			}
+		if !isPrimaryKey(field, primaryKeyOverwrites) {
+			if !isSimpleModelType(fieldType) {
+				if fieldType.Kind() != reflect.Struct {
```

Comment:
Maybe we can reorder this condition, to avoid two ifs with negations. 
```golang
if isPrimaryKey(...) {
...
} else !isSimpleModelType(fieldType) {
...
}
```

Reply:
Actually it can be just:

```golang
if isPrimaryKey(...) {
...
} else fieldType.Kind() == reflect.Struct {
...
}
```

Reply:
Yeas you are right. Done!